### ISSUE TITLE
Screen reader accessibility for ListingCard 

### ISSUE THREAD

**Issue by cliu02** [bloom-housing/ui-components#140]
This issue is from a DAHLIA accessibility audit.

**Description**: When a VoiceOver iOS user accesses the ListingCard component in the DAHLIA directory page (https://dahlia-full.herokuapp.com/listings/for-rent), the content in the data table is not announced. Note that the content is announced as desired with VO macOS and the latest iOS (17.3.1), but not iOS 16.

![Screenshot 2023-11-28 at 1 43 01 PM](https://github.com/bloom-housing/ui-components/assets/32694266/ac023907-5001-4397-96e6-786a0c54f2f3)
